numpy/_core/_add_newdocs.py

```diff
@@ -2943,10 +2943,11 @@
 
 add_newdoc('numpy._core.multiarray', 'ndarray', ('__array__',
     """
-    a.__array__([dtype], /, *, copy=None)
+    a.__array__([dtype], *, copy=None)
 
-    For ``dtype`` parameter it returns either a new reference to self if
-    ``dtype`` is not given or a new array of provided data type if ``dtype``
+    For ``dtype`` parameter it returns a new reference to self if
+    ``dtype`` is not given or it matches array's data type.
+    A new array of provided data type is returned if ``dtype``
     is different from the current data type of the array.
     For ``copy`` parameter it returns a new reference to self if
     ``copy=False`` or ``copy=None`` and copying isn't enforced by ``dtype``
```

Comment:
(not about this line, but can't comment on the lines more below)

Not strictly related to this clarification, but I think for implementors of `__array__` it might be useful to add a link to https://numpy.org/devdocs/user/basics.interoperability.html#the-array-method here. I know this is the docstring for the numpy `__array__` method which has some different semantics (like returning `self`). But when googling for `__array__` you can still end up here.

Reply:
To do this you'll need to add a reference anchor to that subsection of the docs and then reference it here. You should be able to build the docs locally to double check that everything is wired up correctly. Let me know if you need a hand wrangling sphinx.

Reply:
I added a reference anchor - works locally and in the CI artifact.